tutorials/15-external-samplers/external-samplers.jmd

```diff
@@ -0,0 +1,104 @@
+---
+title: Using External Sampler
+permalink: /tutorials/:name/
+mathjax: true
+weave_options:
+  error : false
+---
+
+# Using External Samplers
+
+Turing currently wraps some of the most popular sampling libraries in the Julia ecosystem.
+These wrappers allow new users to seemlessly sample statistical models without leaving Turing.
+However, these wrappers often lack the latests functionalities implemented on the sampling libraries that more experienced users might want to use.
+Moreover, users might want to use samplers currently not wrapped within Turing.
+
+For these reasons, Turing now allows users the possibility of defining their own sampler and using it to sample a Turing model.
+Let us begin by defining a simple Turing model to showcase the use of external samplers.
+For this demonstration we will use a 10 dimensional Neal's funnel:
+
+```julia
+# Import libraries.
+using Turing, Random, LinearAlgebra
+
+d = 10
+@model function funnel()
+    θ ~ Truncated(Normal(0, 3), -3, 3)
+    z ~ MvNormal(zeros(d - 1), exp(θ) * I)
+    return x ~ MvNormal(z, I)
+end
+```
+
+Let us know sample the model to generate some observations and condition our model on them:
+
+```julia
+Random.seed!(1)
```

Comment:
If it's reproducibility we're going for, we might want to use StableRNGs instead?

Reply:
Is this a library?

Reply:
Yep:)

Reply:
I just removed the whole seed since I am not particularly concerned with the exact numbers.